lib/paper_trail/reifier.rb

```diff
@@ -83,6 +88,23 @@ def reify_attributes(model, version, attrs)
         end
       end
 
+      def reify_has_and_belongs_to_many(model, attrs, options)
+        # has_and_belongs_to_many associations are always saved, but if the
+        # option is not set, the associated attributes must be removed so they
+        # are not reified
+        model.class.reflect_on_all_associations(:has_and_belongs_to_many).
+          each { |a|
+            # Don't let habtm associations be reified as attributes
+            ids = attrs.delete("#{a.name.to_s.singularize}_ids")
+            next unless options[:has_and_belongs_to_many]
+            # Update by using `id` IN (...) so if an object has been destroyed
+            # no errors are thrown, although the models association can not be
+            # fully reified (follows standard for has_many & has_many through)
+            model.send(a.name).proxy_association.target =
+              ids.present? ? a.name.to_s.singularize.classify.constantize.where(id: ids).to_a : []
```

Comment:
Won't this load the live (current) records? Shouldn't reification query the `versions` table? What am I missing?